### DISCUSSION TITLE
basic geopandas.GeoDataFrame operations (e.g., buffer, plot) kill python kernel

### DISCUSSION THREAD

**Discussion by erthward** [microsoft/PlanetaryComputer#209] [Q&A]
Hi, All!

I have a body of code I've just been finishing up debugging, but as of a couple days ago I've been having the frustrating experience that simple bits of geopandas code that ran fine before (involving calling the buffer and plot methods of GeoDataFrames, for example) are now killing my Python kernel.  I'm working in the basic Python environment ('CPU - Python') configured on the MSPC Hub. The geodataframe I'm working with is very small (27 rows, 14 columns), but I've also simply made a dummy set of points and tried buffering them, using code that runs fine on my laptop (i.e., `pts = gpd.points_from_xy([5]*3, [1,2,3]); squares = pts.buffer(0.1)`), yet the problem persists.

Things I've tried, to no avail:

  - uninstalling and reinstalling geopandas
  - using bash to search for stranded processes that might be chewing up RAM
  - running the same code either on the main node or in a dask cluster (behavior is the same)

Any ideas what might be wrong? Is it possible that this is just a dependency problem? (I've so far gotten away with being too lazy to use virtual envs, i.e., not wanting to wait eons for conda/mamba, but maybe that would solve my problem?)

Thanks for any input!

Drew

**Comment by TomAugspurger** [comment]
I'm not immediately sure what's going on. 

Just to confirm, you're saying that your `pts = gpd.points_from_xy([5]*3, [1,2,3]); squares = pts.buffer(0.1))` snippet reproduces the problem? I'm not able to reproduce that on the PC Hub:


```python
>>> import geopandas as gpd
>>> pts = gpd.points_from_xy([5]*3, [1,2,3]); squares = pts.buffer(0.1)
>>> print(pts)
<GeometryArray>
[<shapely.geometry.point.Point object at 0x7f7bb7b5af80>,
 <shapely.geometry.point.Point object at 0x7f7bb7b5ae90>,
 <shapely.geometry.point.Point object at 0x7f7bb7b5af80>]
Length: 3, dtype: geometry
```

Does the problem still exist if you stop and restart your notebook server? The python environment that geopandas is installed in is outside of your home directory, and so should be wiped clean each time you restart your notebook server.

**Comment by erthward** [comment]
Hey, Tom.

Thanks! This was super helpful. It made me go back and double-check this minimal example on MSPC, but in a new, empty notebook. Lo and behold, it worked fine for me too! That made me realize that it must be attributable to another package I was loading. I was able to recreate the problem with this minimal example:

```python
>>> import xesmf as xe
>>> import geopandas as gpd
>>> pts = gpd.points_from_xy([5]*3, [1,2,3])
>>> squares = pts.buffer(0.1)
```
thus tracing the problem back to having first loaded the package `xesmf`. I'm not sure what is happening under the hood, but the convenient thing is that I wound up not depending on `xesmf`, so I just removed the package from my notebook and things now seem to be running fine.

Thanks again for taking the time!

Drew
